util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java

```diff
@@ -165,4 +181,70 @@ static URI parseProxyAddress(@Nullable String proxyAddress) {
           String.format("Invalid proxy address '%s'.", proxyAddress), e);
     }
   }
+
+  /** Wrapper class to inject socketKeepAlive property while creating the socket */
+  static class ConfiguredSslSocketFactory extends SSLSocketFactory {
+
+    private final SSLSocketFactory wrappedSockedFactory;
+    private final boolean socketKeepAlive;
+
+    public ConfiguredSslSocketFactory(
+        SSLSocketFactory wrappedSocketFactory, boolean socketKeepAlive) {
```

Comment:
`wrappedSocketFactory` as an argument is not adding any value here, consider initialising to default in the constructor

Reply:
Could we accept socket configuration instead of `socketKeepAlive` ?

Reply:
> `wrappedSocketFactory` as an argument is not adding any value here, consider initialising to default in the constructor